src/model_parsers/TTAParser.cpp

```diff
@@ -198,35 +230,176 @@ std::optional<TTAIR_t::Symbol> TTAParser::ParseSymbolDeclaration(const std::stri
 
 TTA TTAParser::ConvertToModelType(const TTAIR_t &intermediateRep) {
     TTA tta{};
+    for(auto& comp : intermediateRep.components) {
+        auto componentSymbols = ConvertSymbolListToSymbolMap(comp.symbols);
+        tta.symbols.map().insert(componentSymbols.map().begin(), componentSymbols.map().end());
+    }
+    auto internalSymbols = ConvertSymbolListToSymbolMap(intermediateRep.internalSymbols);
+    tta.symbols.map().insert(internalSymbols.map().begin(), internalSymbols.map().end());
+    auto externalSymbols = ConvertSymbolListToSymbolMap(intermediateRep.externalSymbols);
+    tta.InsertExternalSymbols(externalSymbols);
+
     for(auto& comp : intermediateRep.components) {
         tta.components[comp.name] = {
-                .initialLocationIdentifier = comp.initialLocation,
-                .endLocationIdentifier     = comp.endLocation,
+                .initialLocationIdentifier = comp.initialLocation.identifier,
+                .endLocationIdentifier     = comp.endLocation.identifier,
+                .currentLocation           = { comp.initialLocation.isImmediate, comp.initialLocation.identifier },
                 .isMain                    = comp.isMain,
-                .edges                     = ConvertEdgeListToEdgeMap(comp.edges),
-                .symbols                   = ConvertSymbolListToSymbolMap(comp.symbols)
+                .edges                     = ConvertEdgeListToEdgeMap(comp.edges, tta.symbols, comp.name),
         };
     }
     return tta;
 }
 
-std::unordered_map<std::string, TTASymbolType>
-TTAParser::ConvertSymbolListToSymbolMap(const std::vector<TTAIR_t::Symbol> &symbolList) {
-    std::unordered_map<std::string, TTASymbolType> map{};
-    std::for_each(symbolList.begin(), symbolList.end(), [&map] (auto& symbol) { map[symbol.identifier]=symbol.value; });
+TTA::SymbolMap TTAParser::ConvertSymbolListToSymbolMap(const std::vector<TTAIR_t::Symbol> &symbolList) {
+    TTA::SymbolMap map{};
+    std::for_each(symbolList.begin(), symbolList.end(), [&map] (auto& symbol) {
+        std::visit(overload(
+                [&](const int& value)            { map[symbol.identifier] = value; },
+                [&](const float& value)          { map[symbol.identifier] = value; },
+                [&](const bool& value)           { map[symbol.identifier] = value; },
+                [&](const TTATimerSymbol& value) { map[symbol.identifier] = value.current_value; }, // Timers are just read-only floats in the Tick-semantics
```

Comment:
Should timers be read-only floats in the Tick-Semantics? 
No. Tick-steps are able to reset the Timers

Reply:
(Maybe we should do a compile-time check, to check if all timers mentioned in an update is reset to zero, and not something else?)